flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -405,36 +374,13 @@ class JobManager(
 
       currentResourceManager match {
         case Some(rm) =>
-          val future = (rm ? decorateMessage(new RegisterResource(taskManager, msg)))(timeout)
-          future.onComplete {
-            case scala.util.Success(response) =>
-              // the resource manager is available and answered
-              self ! response
-            case scala.util.Failure(t) =>
-              t match {
-                case _: TimeoutException =>
-                  log.info("Attempt to register resource at ResourceManager timed out. Retrying")
-                case _ =>
-                  log.warn("Failure while asking ResourceManager for RegisterResource. Retrying", t)
-              }
-              // slow or unreachable resource manager, register anyway and let the rm reconnect
-              self ! decorateMessage(new RegisterResourceSuccessful(taskManager, msg))
-              self ! decorateMessage(new ReconnectResourceManager(rm))
-          }(context.dispatcher)
-
+          log.info(s"Register task manager $resourceId at the resource manager.")
+          rm ! decorateMessage(new RegisterResource(msg))
```

Comment:
Because it added complexity which is no longer needed. The current design still allows to change it later on. Therefore I don't see a problem in removing it. Just because we might need it in the future, is imho not a good reason to keep unused code around.